From 49b945f9ffa1a803db963bddfd3773b402cb2639 Mon Sep 17 00:00:00 2001
From: Robby Cai <r63905@freescale.com>
Date: Wed, 1 Apr 2015 15:36:37 +0800
Subject: [PATCH] MLK-10530-2 ARM: dts: imx7d-sdb: Move pinmux for sii902x out
 of hoggroup

This pinmux group is also used by EPDC. Thus need to move from hoggroup,
and only set in probe function of its driver. sii902x is already disabled
by default to avoid the conflict, and only enabled in *-hdmi.dts file.

Signed-off-by: Robby Cai <r63905@freescale.com>
---
 arch/arm/boot/dts/imx7d-sdb.dts | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 1237ba9..cd6662f 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -366,6 +366,8 @@
 
 	sii902x@39 {
 		compatible = "SiI,sii902x";
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_sii902x>;
 		interrupt-parent = <&gpio2>;
 		interrupts = <13 IRQ_TYPE_EDGE_FALLING>;
 		mode_str ="1280x720M@60";
@@ -439,7 +441,6 @@
 			fsl,pins = <
 				MX7D_PAD_UART3_CTS_B__GPIO4_IO7	0x14
 				MX7D_PAD_ECSPI2_SS0__GPIO4_IO23	 0x80000000  /* bt reg on */
-				MX7D_PAD_EPDC_DATA13__GPIO2_IO13 0x59
 			>;
 		};
 
@@ -618,6 +619,12 @@
 			>;
 		};
 
+		pinctrl_sii902x: hdmigrp-1 {
+			fsl,pins = <
+				MX7D_PAD_EPDC_DATA13__GPIO2_IO13 0x59
+			>;
+		};
+
 		pinctrl_uart1: uart1grp {
 			fsl,pins = <
 				MX7D_PAD_UART1_TX_DATA__UART1_DCE_TX 0x79
-- 
1.8.0

